src/server/controllers/UserService.ts

```diff
@@ -1,44 +1,64 @@
-import { IUserModel, Users } from '../models/UserModel';
+import { UserModel, Users } from '../models/UserModel';
 import { Logger } from '../util/Logger';
 import * as Promise from 'bluebird';
 import { printAndThrowError } from '../util/common';
+import { can, Action, Target } from '../util/Permissions';
 
 const logger = Logger('controllers/UserService');
 
+export interface User extends UserModel {
+    can(action: Action, target?: Target): boolean
+}
+
+function attachInstanceMethods(user: UserModel | undefined): User | undefined {
```

Comment:
can we do this with prototypes?

`Object.setPrototypeOf(user, { can })` I think would be the way to go about that. Then it's not really a property of the user object itself, but we can still call the function.